### PR TITLE
RPR2 optimizations for viewport

### PR DESCRIPTION
**Pull request by Vsevolod1983**
TCIKET: RPRMAYA-2341

PURPOSE:
Workflow is not optimized for usage of RPR2

EFFECT OF CHANGES:
Workflow is optimized for usage of RPR2
 - Render Callbacks
 - Multithreading processing
 - We update screen on every callback call we get from Core. So, from user perspective, if it drags mouse quite fast only center part of the screen will be updated. (It is slow if update entire screen).

### CODE REVIEW

**Review by bsavery — CHANGES_REQUESTED** [review]
Not a functional change, but please add a bit more detail to the effect description, do you want to wait for a review from @DMotorygin ?

**Comment by Vsevolod1983** [comment]
He had a vacation, but he is back fron today, so assigned

File: FireRender.Maya.Src/NorthStarRenderingHelper.cpp
Comment (Vsevolod1983) [reply FireRender.Maya.Src/NorthStarRenderingHelper.cpp]: Done

File: FireRender.Maya.Src/NorthStarRenderingHelper.cpp
Comment (Vsevolod1983) [reply FireRender.Maya.Src/NorthStarRenderingHelper.cpp]: Done

File: FireRender.Maya.Src/NorthStarRenderingHelper.cpp
Comment (DMotorygin) [inline FireRender.Maya.Src/NorthStarRenderingHelper.cpp:93]: Please remove commented out code

File: FireRender.Maya.Src/Context/TahoeContext.cpp
Comment (DMotorygin) [inline FireRender.Maya.Src/Context/TahoeContext.cpp:481]: Please add new line at the end of file

File: FireRender.Maya.Src/NorthStarRenderingHelper.cpp
Comment (Vsevolod1983) [reply FireRender.Maya.Src/NorthStarRenderingHelper.cpp]: Done

File: FireRender.Maya.Src/Context/TahoeContext.cpp
Comment (Vsevolod1983) [reply FireRender.Maya.Src/Context/TahoeContext.cpp]: Done

**Review by DMotorygin — APPROVED** [review]

**Pull request merged**
